source/fab/util.py

```diff
@@ -50,12 +50,54 @@ def file_walk(path: Path) -> Iterator[Path]:
             yield i
 
 
-@contextmanager
-def time_logger(label):
-    logger.info("\n" + label)
-    start = perf_counter()
-    yield None
-    logger.info(f"{label} took {perf_counter() - start}")
+class Timer(object):
+    """
+    A simple timing context manager.
+    """
+
+    def __init__(self):
+        self._start = None
+        self.taken = None
+
+    def __enter__(self):
+        self._start = perf_counter()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.taken = perf_counter() - self._start
+
+
+class TimerLogger(object):
```

Comment:
Are both `Timer` and `TimerLogger` necessary or is one left over from earlier development? There seems to be duplicated functionality. Could they inherit?